### DISCUSSION TITLE
Channel C++ warnings to Python's warning system

### DISCUSSION THREAD

**Discussion by ischoegl** [Cantera/enhancements#112] [Ideas]
At the moment, the C++ code base issues warning messages via `warn_deprecated` and `warn_user` (the latter introduced in Cantera/cantera#741). In this context, @bryanwweber commented on a new deprecation warning proposed in Cantera/cantera#1084:

> One other comment, and I suppose this applies to all the `warn_deprecated()` calls in C++, I wonder if it's possible to catch those in Python and turn them into real Python `DeprecationWarnings` so they can be controlled by Python's warning interfaces? 

(see [here](https://github.com/Cantera/cantera/pull/1084#issuecomment-901377885); the same presumably applies to `warn_user`).

@speth's response was:

> I think this would require something a bit like how we interface with Python for writing messages to `stdout`, that is, registering a handler in the `Application` class and in that handler making whatever the correct calls are to the Python C API. 

**Additional context:**

* Cantera/cantera#796
* Comment about [Boost logger](https://github.com/Cantera/cantera/pull/741#pullrequestreview-314480685) in Cantera/cantera#741

**Other thoughts:**

As there may be a relatively straight-forward route to this after all, it may make sense to convert this discussion to an enhancement request (i.e. issue).

**Comment by speth** [comment]
I think this resolved by https://github.com/Cantera/cantera/pull/1201, correct?

**Reply by ischoegl** [reply]
Correct. One disadvantage of discussions is that they aren't easily closed.

**Reply by speth** [reply]
Yeah. I'm wondering what happens if we transfer it to an issue and then close it. Mind if I try that here as an experiment?

**Reply by ischoegl** [reply]
Feel free!

**Reply by speth** [reply]
Ah, no such luck -- I think this would just create yet another Issue. I'm going to try another option, which is to assign a different category.

**Reply by ischoegl** [reply]
just marked it as completed on this end.

**Reply by speth** [reply]
Yes. To clarify, that's a category I just created for this purpose.

**Reply by ischoegl** [reply]
:+1: Great!

**Comment by speth** [comment]
Apparently discussions gained a "Close" feature at some point, so now we don't have to rely on categories as an indicator.
